Update and restructure PostgreSQL integration docs.

alistairmatthews opened the pull request:
The existing PostgreSQL integration document is comprehensive but difficult for beginners. This PR is intended to keep the comprehensive information but make it easier for those who want to get going quickly.

This PR:

- Splits the large integration doc into two: Hosting and Client.
- Adds a getting started short tutorial.
- Adds a getting started tutorial for both hosting and client integration.
- Adds a **Properties of the PostgreSQL resources** section to the client reference docs. This describes what configuration values are injected by the `WithReference()` method. I've put this in the Client doc, where you might use them, but added a Note and a link in the Hosting doc.

Addresses: #30, #9

IEvangelist commented:
This is a good start, let's make the following updates:

- The "get started" article per integration should be simplified so that it's just the brief intro, hosting integration package install instructions, the primary API to add the resource, link to advanced configs and scenarios (hosting reference it's called now), details of what config is injected, and then a client section that has a language pivot. C#, Python, and JS...

No need for the client standalone article.

alistairmatthews commented:
- Pivots for Python and JavaScript completed.
- Complete, basic examples are included for both of these.
- I've kept the separate client integration article. Fowler was in favour.
- I've kept the screenshots at the end. Some people didn't like them but, imho, they provide a good payoff.

@IEvangelist, @davidfowl could you see what you think? Ready to propagate this structure to the other integrations when you're happy. Thanks!

IEvangelist commented:
This is looking good! A few nits, set page navigation limits:

- The postgres-get-started.mdx should have frontmatter of `prev: false` to avoid this:
  
  <img width="1449" height="253" alt="image" src="https://github.com/user-attachments/assets/62dc90f4-491a-45a5-aa8c-6f0c44e580cb" />

- Likewise, the postgresql-extensions.mdx should set `next: false` to avoid this

  <img width="1430" height="217" alt="image" src="https://github.com/user-attachments/assets/dc5e4965-9152-45b1-a855-472b06d5dfe2" />

In the postgres-get-started.mdx, I see that the sidebar has headings for all languages...instead they should be generic-shared headings with the pivot content under each section. In other words, no <Pivot></Pivot> should ever contain a heading, as they're conditionally rendered and we don't update the sidebar (although, that might change).

Today, for example, if you have C# selected and click on the Node.js or Python headings here - they do nothing.

<img width="440" height="509" alt="image" src="https://github.com/user-attachments/assets/439c4006-e96e-4628-b434-844d04ce7705" />


Also, I thought we were not going to have the running solution section with the dashboard screenshots. Too much to maintain, and not really useful as the get started is only conceptual in nature, not really giving users enough code to have a fully running example as depicted.

Finally, the Next steps section, the titles of the link cards are too long. The description should have the details of what they're clicking into and the title should be shorter to avoid tiny right-arrow SVGs:

<img width="1417" height="505" alt="image" src="https://github.com/user-attachments/assets/17773608-2ab2-4381-bec1-9d689e95289c" />

alistairmatthews commented:
> This is looking good! A few nits, set page navigation limits:
> 
> * The postgres-get-started.mdx should have frontmatter of `prev: false` to avoid this:
>   <img alt="image" width="1449" height="253" src="https://private-user-images.githubusercontent.com/7679720/521418094-62dc90f4-491a-45a5-aa8c-6f0c44e580cb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2OTM3NzAsIm5iZiI6MTc2NDY5MzQ3MCwicGF0aCI6Ii83Njc5NzIwLzUyMTQxODA5NC02MmRjOTBmNC00OTFhLTQ1YTUtYWE4Yy02ZjBjNDRlNTgwY2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMDJUMTYzNzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDhhYmJkMjE3Yjg5MzM4MjMyMDRhZjQ1MWUyNjRkYzQyZmMwMTdmNmEwMjk0NGQwMTMxNzE0YjZmZmRiNzk2NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FV4JY2xYsg9CHXgTpCvjcbXdYYjfKNo8fe6AgqfPn58">
> * Likewise, the postgresql-extensions.mdx should set `next: false` to avoid this
>   <img alt="image" width="1430" height="217" src="https://private-user-images.githubusercontent.com/7679720/521418632-dc5e4965-9152-45b1-a855-472b06d5dfe2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2OTM3NzAsIm5iZiI6MTc2NDY5MzQ3MCwicGF0aCI6Ii83Njc5NzIwLzUyMTQxODYzMi1kYzVlNDk2NS05MTUyLTQ1YjEtYTg1NS00NzJiMDZkNWRmZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMDJUMTYzNzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDdhNDBlNjAyODgwMTA2YWY2OTJjYThmZTJmMjk2MmJkMWVlMmU5MDIzYzNiMGU1OGQzZjMzNGJmNDcwZjIwNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6KuHHo1xWkw4hJNNCKeGYucDfM2hz9REikDucdwKP0A">
> 

Thanks. Fixed those and put them on my checklist for future articles.

> In the postgres-get-started.mdx, I see that the sidebar has headings for all languages...instead they should be generic-shared headings with the pivot content under each section. In other words, no should ever contain a heading, as they're conditionally rendered and we don't update the sidebar (although, that might change).
> 
> Today, for example, if you have C# selected and click on the Node.js or Python headings here - they do nothing.
> 
> <img alt="image" width="440" height="509" src="https://private-user-images.githubusercontent.com/7679720/521420121-439c4006-e96e-4628-b434-844d04ce7705.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2OTM3NzAsIm5iZiI6MTc2NDY5MzQ3MCwicGF0aCI6Ii83Njc5NzIwLzUyMTQyMDEyMS00MzljNDAwNi1lOTZlLTQ2MjgtYjQzNC04NDRkMDRjZTc3MDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMDJUMTYzNzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWJhZDM1OTM1ODk1YWZiN2Q2MDZiYTdjZDE5MDUyMjExMWYzNjhjYzczZDExYmMyOWMwMzk4MmRkYjFiZTY2YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.26lCjKpaK-MMVCRtPZC8LEktU1IBqVCgQxQVQipMT-g">
> Also, I thought we were not going to have the running solution section with the dashboard screenshots. Too much to maintain, and not really useful as the get started is only conceptual in nature, not really giving users enough code to have a fully running example as depicted.
> 

OK, thanks. I'll have to adjust the titles so they work for all three languages. This one isn't fixed yet.

> Finally, the Next steps section, the titles of the link cards are too long. The description should have the details of what they're clicking into and the title should be shorter to avoid tiny right-arrow SVGs:
> 
> <img alt="image" width="1417" height="505" src="https://private-user-images.githubusercontent.com/7679720/521422146-17773608-2ab2-4381-bec1-9d689e95289c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ2OTM3NzAsIm5iZiI6MTc2NDY5MzQ3MCwicGF0aCI6Ii83Njc5NzIwLzUyMTQyMjE0Ni0xNzc3MzYwOC0yYWIyLTQzODEtYmVjMS05ZDY4OWU5NTI4OWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMDJUMTYzNzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGY0NWFlYzEwOWExZjBiN2I2ZWY1ZTY3YWY1MGU4ZTJiMTgxYzQxNDYzNTY4ZjllZjIyNzc2NjJmNDY5MTlmMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.RKgOHjqnFP2DNyt2aQD8NPlWvuBPeACx9nDsJsCQGAw">

Fixed!
